src/core/qgsabstractdatabaseproviderconnection.h

```diff
@@ -0,0 +1,406 @@
+#include "qgis_core.h"
+#include "qgsfields.h"
+#include "qgsexception.h"
+
+#include <QObject>
+
+/**
+ * The QgsAbstractDatabaseProviderConnection class provides common functionality
+ * for DB based connections.
+ *
+ * This class performs low level DB operations without asking
+ * the user for confirmation or handling currently opened layers and the registry
+ * entries, it is responsibility of the client code to keep layers in sync.
+ * The class methods will throw exceptions in case the requested operation
+ * is not supported or cannot be performed without errors.
+ *
+ * \ingroup core
+ * \since QGIS 3.10
+ */
+class CORE_EXPORT QgsAbstractDatabaseProviderConnection : public QgsAbstractProviderConnection
+{
+
+    Q_GADGET
+
+  public:
+
+    /**
+     * Flags for table properties.
+     *
+     * Flags can be useful for filtering the tables returned
+     * from tables().
+     */
+    enum TableFlag
+    {
+      Aspatial = 1 << 1,          //!< Aspatial table (it does not contain any geometry column)
+      Vector = 1 << 2,            //!< Vector table (it does contain one geometry column)
+      Raster = 1 << 3,            //!< Raster table
+      View = 1 << 4,              //!< View table
+      MaterializedView = 1 << 5,  //!< Materialized view table
+    };
+
+    Q_ENUMS( TableFlag )
+    Q_DECLARE_FLAGS( TableFlags, TableFlag )
+    Q_FLAG( TableFlags )
+
+    /**
+     * The TableProperty class represents a database table or view.
+     *
+     * In case the table is a vector spatial table and it has multiple
+     * geometry columns, separate entries for each geometry column must
+     * be created.
+     *
+     * In case the table is a vector spatial table and the geometry column
+     * can contain multiple geometry types and/or SRIDs, a clone of the property
+     * for the individual geometry type/SRID can be retrieved with at(i)
+     */
+    struct TableProperty
+    {
+
+#ifdef SIP_RUN
+        SIP_PYOBJECT __repr__();
+        % MethodCode
+        QString str = QStringLiteral( "<QgsAbstractDatabaseProviderConnection.TableProperty: '%1'>" ).arg( sipCpp->tableName() );
+        sipRes = PyUnicode_FromString( str.toUtf8().constData() );
+        % End
+#endif
+
+      public:
+
+        /**
+         * Returns the table name
+         * \see defaultName()
+         */
+        QString tableName() const;
+
+        /**
+         * Sets the table name to \a name
+         * \see defaultName()
+         */
+        void setTableName( const QString &name );
+
+        /**
+         * Appends the geometry column \a type with the given \a srid to the geometry column types list
+         */
+        void addGeometryType( const QgsWkbTypes::Type &type, const int srid );
+
+        /**
+         * Returns the list of geometry column types and SRIDs.
+         * The method returns a list of tuples (QgsWkbTypes::Type, SRID)
+         */
+        QList<QPair<int, int>> geometryTypes() const;
```

Comment:
SRIDs are evil -- don't use them. They'll be gone in QGIS 4.0.

Always use auth:id strings instead. See https://github.com/qgis/qgis4.0_api/issues/94.

Reply:
Are you sure I should do this switch now?
Note that this implementation is modeled on PG provider, and that type/SRID information comes directly from the provider, I don't think I should change the provider code now.

Reply:
Well, I don't want to see srids used in any new stable API. I don't think you should modify the actual backend provider code, but we should avoid this being public and stable API.